packages/block-library/src/search/index.php

```diff
@@ -188,25 +188,47 @@ function styles_for_block_core_search( $attributes ) {
 	$has_border_radius = ! empty( $attributes['style']['border']['radius'] );
 
 	if ( $has_border_radius ) {
-		// Shared style for button and input radius values.
+		$default_padding = '4px';
 		$border_radius   = $attributes['style']['border']['radius'];
-		$border_radius   = is_numeric( $border_radius ) ? $border_radius . 'px' : $border_radius;
-		$shared_styles[] = sprintf( 'border-radius: %s;', esc_attr( $border_radius ) );
-
-		// Apply wrapper border radius if button placed inside.
-		$button_inside = ! empty( $attributes['buttonPosition'] ) &&
+		$button_inside   = ! empty( $attributes['buttonPosition'] ) &&
 			'button-inside' === $attributes['buttonPosition'];
 
-		if ( $button_inside ) {
-			// We adjust the border radius value for the outer wrapper element
-			// to make it visually consistent with the radius applied to inner
-			// elements. calc() is used to support non-pixel CSS units.
-			$default_padding  = '4px';
-			$wrapper_styles[] = sprintf(
-				'border-radius: calc(%s + %s);',
-				esc_attr( $border_radius ),
-				esc_attr( $default_padding )
-			);
+		if ( is_array( $border_radius ) ) {
+			// Apply styles for individual corner border radii.
+			foreach ( $border_radius as $key => $value ) {
+				if ( null !== $value ) {
+					// Convert camelCase key to kebab-case.
+					$name = strtolower( preg_replace( '/(?<!^)[A-Z]/', '-$0', $key ) );
```

Comment:
Do we need to sanitize or escape the key as well as converting to kebab-case? We're wrapping the  `$value` in an `esc_attr` call below, so I was wondering if we should either add another `esc_attr` call for the `$name` variable, or make sure that the `$key` has been stripped of unexpected characters?

Reply:
I was thinking that the keys were only the properties we set e.g. `topLeft`, `topRight`, `bottomLeft`, `bottomRight`. It could be possible to fudge that attribute though so I'll fix that up. Better safe than sorry in any case!

Reply:
✅ Done